backend/server/routers/courses.py

```diff
@@ -297,9 +309,17 @@ def coursesUnlockedWhenTaken(userData: UserData, courseToBeTaken: str):
     user.add_courses({courseToBeTaken: [getCourse(courseToBeTaken)['UOC'], None]})
     ## final state
     courses_now_unlocked = unlocked_set(getAllUnlocked(user)['courses_state'])
+    new_courses = courses_now_unlocked - courses_initially_unlocked
+
+    ## Differentiate direct and indirect unlocks
+    path_to = set(getCourse(courseToBeTaken)['path_to'])
+    direct_unlock = new_courses.intersection(path_to)
+    indirect_unlock = new_courses - direct_unlock
 
     return {
         'courses_unlocked_when_taken' : sorted(list(courses_now_unlocked - courses_initially_unlocked)),
```

Comment:
1. this can just be `new_courses`
2. what's the reason for returning `courses_unlocked_when_taken` when it appears to be just a duplicate of `direct_unlock + indirect_unlock`?

Reply:
Had it initially to maintain compatability with FE but you're right, not needed anymore.